### PR TITLE
Register types for loads of outer blocks

### PR DESCRIPTION
**Pull request by kasperl**

### CODE REVIEW

File: src/compiler/propagation/type_propagator.cc [hunk 1]
```diff
@@ -254,7 +254,7 @@ void TypePropagator::propagate(TypeDatabase* types) {
   sites_.for_each([&](uint8* site, Set<TypeVariable*>& results) {
     type.clear(words_per_type());
     for (auto it = results.begin(); it != results.end(); it++) {
-      type.add_all((*it)->type(), words_per_type());
+      type.add_all_even_blocks((*it)->type(), words_per_type());
```
Comment (floitsch) [inline src/compiler/propagation/type_propagator.cc:257]: It took me way too long to realize this is not talking about "even blocks", where "even" means not odd.
```suggestion
      type.add_all_also_blocks((*it)->type(), words_per_type());
```

File: src/compiler/propagation/type_set.h [hunk 2]
```diff
@@ -53,7 +53,7 @@ class TypeSet {
     unsigned next() {
       ASSERT(has_next());
       uword bits = bits_;
-      // Compute the index of the next bit by counting 
+      // Compute the index of the next bit by counting
```
Comment (floitsch) [inline src/compiler/propagation/type_set.h:56]: not important, but the comment should fit on one line. no?

**Review by floitsch — APPROVED** [review]
LGTM.

File: src/compiler/propagation/type_propagator.cc (hunk 1 above)
Comment (kasperl) [reply src/compiler/propagation/type_propagator.cc:257]: Done.

**Pull request merged**